### PR TITLE
feat: Manual `Propagator` initialization and `AnySurfaceReached` aborter

### PR DESCRIPTION
**Pull request by andiwand**
This allows to manually initialize the propagation so `propagate(state)` can be called multiple times without destruction of the state in-between. I also added an `AnySurfaceReached` aborter which, in combination with the other change, allows users to jump from surface to surface using the same propagator state without any resets.

### CODE REVIEW

File: Core/include/Acts/Propagator/Propagator.hpp
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/Propagator.hpp:470]: This did just move the two `makeState` functions to another place?

File: Core/include/Acts/Propagator/Propagator.hpp
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/Propagator.hpp:482]: As discussed, we could hide this somehow in `makeState`

File: Core/include/Acts/Propagator/StandardAborters.hpp
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/StandardAborters.hpp:202]: Why not `targetSurface`? Maybe comment...

File: Core/include/Acts/Propagator/StandardAborters.hpp
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/StandardAborters.hpp:189]: Small doxygen comment maybe

**Review by benjaminhuth — COMMENTED** [review]
Some comments

File: Tests/UnitTests/Core/Navigation/DetectorNavigatorTests.cpp
Comment (andiwand) [inline Tests/UnitTests/Core/Navigation/DetectorNavigatorTests.cpp:1]: @ssdetlab I had to modify this test a bit to get my changes working

My thought was here that the state is broken after the exception happened anyway and none of the navigation functions should be called afterwards by the `Propagator`. What do you think?

**Review by benjaminhuth — APPROVED** [review]
lgtm, I do not yet start merging to wait for @ssdetlab approval

**Comment by ssdetlab** [comment]
Hi @andiwand, @benjaminhuth. The changes look reasonable to me. Maybe add a small comment clarifying that the `makeState` call is going to initialize the navigator, but I don't insist.

**Review by benjaminhuth — APPROVED** [review]
Let's get it in!

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [a48fa8ac4d74f6647c64bb97e14d6d21276636f1]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7474623)!

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482634)||
|:red_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482633)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482632)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482631)||
|:red_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482630)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482629)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482628)||
|:red_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482627)||
|:red_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482626)||
|:red_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482625)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482624)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482623)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482622)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482621)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482620)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482619)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482618)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482617)||
|:red_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482616)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482615)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482614)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482613)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482612)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482611)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482610)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482609)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482608)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482607)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482606)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39482605)||

**Pull request merged**
